big_tests/tests/privacy_SUITE.erl

```diff
@@ -875,23 +876,32 @@ subscribe(Who, Whom) ->
     escalus:send(Whom, escalus_stanza:roster_add_contact(Who,
                                                         [<<"enemies">>],
                                                         <<"Enemy1">>)),
+    escalus:assert(is_iq_result, escalus:wait_for_stanza(Whom)),
     PushReq2 = escalus:wait_for_stanza(Whom),
     escalus:assert(is_roster_set, PushReq2),
     escalus:send(Whom, escalus_stanza:iq_result(PushReq2)),
-    escalus:assert(is_iq_result, escalus:wait_for_stanza(Whom)),
 
     %% 'Whom' sends subscribed presence
     escalus:send(Whom, escalus_stanza:presence_direct(
                          escalus_client:short_jid(Who), <<"subscribed">>)),
 
     %% 'Who' receives subscribed
-    Stanzas = escalus:wait_for_stanzas(Who, 2),
+    Stanzas = escalus:wait_for_stanzas(Who, 3),
 
-    check_subscription_stanzas(Stanzas, <<"subscribed">>),
-    escalus:assert(is_presence, escalus:wait_for_stanza(Who)).
+    check_subscription_stanzas(Stanzas, <<"subscribed">>, <<"available">>),
+    ok.
 
 check_subscription_stanzas(Stanzas, Type) ->
+    check_subscription_stanzas(Stanzas, Type, none).
+
+check_subscription_stanzas(Stanzas, Type, PresenceType) ->
```

Comment:
Is it possible to unify the `check_subscription_stanzas` function from this and `presence_SUITE`?